### PR TITLE
Upload Documents Automation and Refactoring

### PR DESCRIPTION
**Pull request by SharifChoudhury1**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 1]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
-    await I.click("//div[text()='Documents']");
-    await I.see('fileUpload.txt');
+    //Start of Upload document page
+    I.see('Upload Document');
+    I.see('Case Number:');
+    I.see('Case documentation (Optional)');
+    I.see('Add new');
+    I.see('Upload documentation for the case');
+
+    I.click("//button[@class='button write-collection-add-item__top']");
```
Comment (pats-john) [inline src/test/end-to-end/pages/uploadDocument/uploadDocument.js:14]: Sometimes it's more readable to use Buttons Names with Clear Text than give XPath Locators.

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 2]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
+    I.see('Case documentation');
+    I.see('Type of Document (Optional)'); 
+    I.see('Document (Optional)');      
+    I.attachFile('#documentCollection_0_uploadedDocument', 'data/fileUpload.txt');
+    I.wait(5);
```
Comment (pats-john) [inline src/test/end-to-end/pages/uploadDocument/uploadDocument.js:19]: Replace this hard wait with a  waitForText in the next line...

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 3]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
+    I.see('Short Description (Optional)');
+    I.fillField('#documentCollection_0_shortDescription', commonConfig.shortDescription);
+
+    I.click(commonConfig.continue);
```
Comment (pats-john) [inline src/test/end-to-end/pages/uploadDocument/uploadDocument.js:23]: Is there no text of heading that you can verify between the Continue and the Submit....

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 4]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
+    I.click("//button[@class='button']");
+    I.seeElement("//div[@class='alert-message']");
+    I.wait(5);
```
Comment (pats-john) [inline src/test/end-to-end/pages/uploadDocument/uploadDocument.js:26]: Why the hard wait...?

**Review by pats-john — CHANGES_REQUESTED** [review]

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 5]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
+    I.see('Add new');
+    I.see('Upload documentation for the case');
+
+    I.click("//button[@class='button write-collection-add-item__top']");
+    I.see('Case documentation');
+    I.see('Type of Document (Optional)'); 
+    I.see('Document (Optional)');      
+    I.attachFile('#documentCollection_0_uploadedDocument', 'data/fileUpload.txt');
+    I.wait(5);
+    I.see('Short Description (Optional)');
```
Comment (pats-john) [inline src/test/end-to-end/pages/uploadDocument/uploadDocument.js:20]: Bunch the I.see text together...

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 6]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
+    I.see('Upload Document');
+    I.see('Case Number:');
+    I.see('Case documentation (Optional)');
+    I.see('Add new');
+    I.see('Upload documentation for the case');
+
+    I.click("//button[@class='button write-collection-add-item__top']");
+    I.see('Case documentation');
+    I.see('Type of Document (Optional)'); 
+    I.see('Document (Optional)');
```
Comment (pats-john) [inline src/test/end-to-end/pages/uploadDocument/uploadDocument.js:17]: Select a document type on the select option

**Review by pats-john — CHANGES_REQUESTED** [review]

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 7]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
+    I.see('Case documentation');
+    I.see('Type of Document (Optional)'); 
+    I.see('Document (Optional)');      
+    I.attachFile('#documentCollection_0_uploadedDocument', 'data/fileUpload.txt');
+    I.wait(5);
+    I.see('Short Description (Optional)');
+    I.fillField('#documentCollection_0_shortDescription', commonConfig.shortDescription);
+
+    I.click(commonConfig.continue);
+    I.click("//button[@class='button']");
```
Comment (dharrisMOJ) [inline src/test/end-to-end/pages/uploadDocument/uploadDocument.js:24]: Consider using buttons names with clear text over XPath Locators as mentioned earlier. We have a commonConfig file that can be used for the submit button.

**Review by dharrisMOJ — CHANGES_REQUESTED** [review]

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js (hunk 7 above)
Comment (SharifChoudhury1) [reply src/test/end-to-end/pages/uploadDocument/uploadDocument.js:24]: Added I.click(commonConfig.submit);

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js [hunk 8]
```diff
@@ -4,14 +4,32 @@ const commonConfig = require('../../data/commonConfig.json');
...
+    I.seeElement("//div[@class='alert-message']");
```
[the remark this answers is not in the document]
Comment (SharifChoudhury1) [reply src/test/end-to-end/pages/uploadDocument/uploadDocument.js:25]: Verified text with I.see('has been updated with event: Upload Document');

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js (hunk 6 above)
Comment (SharifChoudhury1) [reply src/test/end-to-end/pages/uploadDocument/uploadDocument.js:17]: I.selectOption('Type of Document (Optional)', 'ET1');

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js (hunk 5 above)
Comment (SharifChoudhury1) [reply src/test/end-to-end/pages/uploadDocument/uploadDocument.js:20]: Bunched I.see's together

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js (hunk 3 above)
Comment (SharifChoudhury1) [reply src/test/end-to-end/pages/uploadDocument/uploadDocument.js:23]: Verified 'Upload Documents' and 'Case Number:'

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js (hunk 2 above)
Comment (SharifChoudhury1) [reply src/test/end-to-end/pages/uploadDocument/uploadDocument.js:19]: Removed hard waits.

File: src/test/end-to-end/pages/uploadDocument/uploadDocument.js (hunk 1 above)
Comment (SharifChoudhury1) [reply src/test/end-to-end/pages/uploadDocument/uploadDocument.js:14]: Used button name with clear text

**Review by pats-john — APPROVED** [review]

**Pull request merged**